src/main/java/com/epam/task6/logic/DAO/BookDAO.java

```
@@ -0,0 +1,64 @@
+package com.epam.task6.logic.DAO;
+
+import com.epam.task6.logic.Field;
+import com.epam.task6.logic.search.SearchBook;
+import com.epam.task6.logic.search.factory.CreateSearch;
+import com.epam.task6.logic.search.factory.SearchBookFactory;
+import com.epam.task6.logic.sort.SortBook;
+import com.epam.task6.logic.sort.factory.CreateSort;
+import com.epam.task6.logic.sort.factory.SortBookFactory;
+import com.epam.task6.model.Book;
+import org.apache.log4j.Logger;
+
+import java.util.List;
+
+
+public class BookDAO implements DAO<Book> {
+
+    private static final Logger log = Logger.getLogger(BookDAO.class);
+    private List<Book> books;
+
+    public BookDAO(List<Book> books) {
+
+        this.books = books;
+    }
+
+    public List<Book> getBooks() {
+
+        return books;
+    }
+
+    @Override
+    public List<Book> getAll() {
+
+        return books;
+    }
+
+    @Override
+    public void add(Book book) {
+
+        books.add(book);
```

according to the task you need to throw an exception if already present